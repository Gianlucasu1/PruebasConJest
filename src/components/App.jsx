import React from 'react'

export const App = () => {
    return (
        <div>
            <h1>Hello Gianluca</h1>
            <p>Hellos</p>
        </div>
    )
}
