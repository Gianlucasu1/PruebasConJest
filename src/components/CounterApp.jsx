import PropTypes from "prop-types";
import {useState} from "react"

//Esta funciÃ³n esta afuera del counter app ya que no necesita del valor de value que le pasamos como propiedad al counter app
// Y por eso no es necesario incluirla cada vez que el CounterApp se renderize

//const handleEvent = (eventNew) => {
  //  console.log(eventNew);
//}


export const CounterApp = ({value}) => {

    const [number,setNumber]=useState(value);

    const handleAdd = (event) => {
        console.log(event)
        setNumber(x=>x+1)
    }

    const handleSubstract = () => {
        if(number == 0){
            return 0
        }
        else{
        setNumber(x => x-1);
        }
    }

    const handleReset = (  ) => setNumber( x => { return x = 0 + value}) ;

    return (
        <>
        <h1>Counter App</h1>
        <h2 data-testid="counter-test">This is the counter Value : {number}</h2>
        <div className="center">
            <button className="btn-counter" onClick={handleAdd}>+1</button>
            <button className="btn-counter" onClick={handleSubstract}>-1</button>
            <button arial-label="btn-reset" className="btn-counter" onClick={handleReset}>RESET</button>
        </div>
        </>
    )
}

CounterApp.propTypes={
    value: PropTypes.number.isRequired
}
CounterApp.defaultProps = {
    value : 0
}