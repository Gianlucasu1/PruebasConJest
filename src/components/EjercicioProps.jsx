import React from "react";
// We need to install proptypes in the main carpet using npm install prop-types
import PropTypes from "prop-types";

export const EjercicioProps = ({name, lastname, number}) => {

    


    return (
    <>

    <h1 data-testid="test-tittle">Hola mi nombre es {name}</h1>
    <p data-testid="test-p">{lastname}</p>
    <p>My favorite number is {number}</p>

    </>
    );
}


EjercicioProps.propTypes = {
    name: PropTypes.string.isRequired,
    number: PropTypes.number.isRequired,
    lastname: PropTypes.string.isRequired
}

EjercicioProps.defaultProps = {
    number:8
}