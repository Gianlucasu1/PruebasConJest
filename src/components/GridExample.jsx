import React from "react"

export const GridExample = () => {
    return (

        
            <div className="grid">
                 <div className="grid_2">
                     <div>testing</div>
                     <div>testing</div>
                     <div>testing</div>
                     <div>testing</div>
                     <div>testing</div>
                     <div>testing</div>
                 </div>
                 <div>Voluptatum ullam sunt sequi molestiae. Magnam voluptatum temporibus, neque quibusdam omnis aliquid ex aspernatur at excepturi facilis. Quia commodi mollitia, necessitatibus et error officia, eveniet, soluta veniam facilis deserunt assumenda.</div>
                 <div>Aliquam doloremque cum ratione blanditiis quam voluptas mollitia temporibus nobis suscipit molestiae. Modi, iusto obcaecati corrupti sed id dignissimos est possimus consequatur, voluptatum nihil praesentium maxime vero fugit, ea ab.</div>
                 <div className="element4">Quidem doloribus minus beatae porro, minima natus, repudiandae explicabo sit impedit ab, non nobis rem praesentium corporis est quaerat optio. Commodi eum minus minima doloribus beatae ipsum consectetur, quas molestiae.</div>
                 <div>Quibusdam minima mollitia accusantium. Non debitis alias exercitationem commodi magni ad ut ipsum recusandae odio tempora neque sit, quae fugiat voluptatibus consequatur quod vero corrupti rerum, delectus tempore nihil minus.</div>
                 <div>Cumque at itaque corporis id magnam placeat vero, veniam, vel quas iusto repellendus reiciendis. Dolore totam dolorum exercitationem blanditiis dolor consequatur commodi officia! Ipsum magnam voluptates facilis illo similique officia.</div>
                 <div>Natus nihil ullam maxime sed ratione repellat doloribus, culpa magni laudantium quis quod officia, voluptate error officiis corrupti fugit! Ducimus, quo. Vero harum debitis reiciendis deserunt consequatur nobis quisquam soluta.</div>
                 <div>Excepturi minus ipsam quos veniam laborum deserunt, earum ab vel quibusdam harum adipisci dolorum, a aspernatur non porro nam aliquid fuga. Vitae sequi aliquid voluptatum recusandae dolore culpa excepturi. Quidem?</div>
                 <div>Placeat facere et quo porro provident velit molestias inventore quas consequuntur culpa dolore, eligendi neque in possimus quam tenetur iure esse autem ducimus architecto corrupti. Laudantium sit delectus illum earum.</div>
                 <div>Animi deserunt est tempora assumenda ipsa nesciunt officia dolor recusandae quidem explicabo, distinctio vero quibusdam harum laboriosam itaque! Neque distinctio dolorem unde, odio ducimus numquam aliquam quaerat ipsa quam dolor!</div>
            </div>
        

    )
}