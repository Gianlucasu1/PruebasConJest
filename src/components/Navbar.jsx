import React from 'react'
import pic1 from "../images/react.png"

export const Navbar = () => {
    return (
        <header>
            <a className="logo">Image</a>
            <nav>
                <ul className="nav__links">
                    <li><a href="#">Home</a></li>
                    <li><a href="#">Info</a></li>
                    <li><a href="#">About us</a></li>
                </ul>
            </nav>
            <a href="#" className="cta"><button>Contact</button></a>
        </header>
    )
}
