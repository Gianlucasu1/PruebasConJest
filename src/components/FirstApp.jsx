import { useState } from "react";
import { App } from "./App";




export const FirstApp = () => {

    const [counter, setCounter] = useState(0);
    const [impressions, setImpresions] = useState([]);






    let miNombre = "Gianluca";

    const myFunction = (()=> {

        console.log("The function worked");
        setImpresions(x =>
                [...x, <App/>]
            )
        
    });
    

    return (
        <>  
            <section className="section_1">
                <h1>First app done, congratulations {miNombre}</h1>
                <button onClick={myFunction}>Click here</button>
            </section>
            <div className="section_2">
                {impressions}
            </div>                
        </>
    )
}
