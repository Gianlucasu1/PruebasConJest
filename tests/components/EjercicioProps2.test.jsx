/**
 * @jest-environment jsdom
 */

import React from "react"
import { render, screen } from "@testing-library/react";
import { EjercicioProps } from "../../src/components/EjercicioProps"

describe('Pruebas en <EjercicioProps />', () => {

    const name = "Gianluca";

    test('Debe hacer match con el snapshot', () => {

        const {container} = render(<EjercicioProps name={name} lastname="Gianluca" />)

        expect( container ).toMatchSnapshot();

    })

    test("Debe mostrar Hola Gianluca", () =>{
        render(<EjercicioProps name={name} lastname="Gianluca" />)
        screen.debug();
    })
    
});
