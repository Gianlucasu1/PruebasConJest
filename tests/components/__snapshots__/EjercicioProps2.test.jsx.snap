// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <EjercicioProps /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-tittle"
  >
    Hola mi nombre es 
    Gianluca
  </h1>
  <p
    data-testid="test-p"
  >
    Gianluca
  </p>
  <p>
    My favorite number is 
    8
  </p>
</div>
`;
